### PR TITLE
ref(ui): Extract useOwner{s,Options}

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This hook may be used to load member and team options for a select component.

This is a spin off of https://github.com/getsentry/sentry/pull/69269

### CODE REVIEW

File: static/app/utils/useOwners.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    [currentValue]
+  );
+  useMembers({ids: ensureUserIds});
+
+  const {
+    members,
+    fetching: fetchingMembers,
+    onSearch: onMemberSearch,
+    loadMore: loadMoreMembers,
+  } = useMembers();
```
Comment (ryan953) [inline static/app/utils/useOwners.tsx:32]: why does the hook need to be listed twice?

Comment (evanpurkhiser) [reply static/app/utils/useOwners.tsx:32]: lol good question. This is how it is in the other component. Let me see if this is not required

File: static/app/utils/useOwners.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+/**
+ * Hook to fetch owner options
+ */
+export function useOwners({currentValue}: Options) {
+  // Ensure the current value of the fields members is loaded
+  const ensureUserIds = useMemo(
+    () =>
+      currentValue?.filter(item => item.startsWith('user:')).map(user => user.slice(7)),
```
Comment (ryan953) [inline static/app/utils/useOwners.tsx:22]: `slice(7)`? what are we trying to do, maybe a `replace()` call would be clearer

**Review by ryan953 — APPROVED** [review]
not too many useMemo or useCallbacks inside of useOwnerOptions, but seems fine to get started with

File: static/app/utils/useOwners.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/utils/useOwners.tsx:22]: we would probably want to use a regex replace to make sure it trims from the fromt, but I think this is clear enough IMO

Comment (ryan953) [reply static/app/utils/useOwners.tsx:22]: the mysterious part is that `user:` is only 5 chars, but we're stripping off 7.

Comment (evanpurkhiser) [reply static/app/utils/useOwners.tsx:22]: 👀 that sounds like a ream bug lol

**Pull request merged**